dev/tests/integration/framework/Magento/TestFramework/MessageQueue/PublisherConsumerController.php

```diff
@@ -80,6 +80,11 @@ public function initialize()
                 "This test relies on *nix shell and should be skipped in Windows environment."
             );
         }
+        if (!$this->amqpHelper->isAvailable()) {
+            throw new EnvironmentPreconditionException(
+                "This test relies on RabbitMQ Management Plugin."
```

Comment:
I'm a bit concerned about such approach. When such plugin is actually not available? How can we ensure AMQP helper did not became not available accidentally?

Reply:
The management plugin is distributed with RabbitMQ, but needs to be explicitly enabled. Should we throw a different exception to so the test fails with the same error message. My primary concern was the undocumented dependency on the management plugin.

Reply:
@pmclain,

> Like any other plugin it must be enabled before it can be used

Not enabled by default, right? Is there any issue with enabling such plugin whenever you notice a failing test on your local env?

Reply:
None at all and I agree with your original assessment. I actually updated this to throw `\Magento\TestFramework\MessageQueue\PreconditionFailedException` which will fail the test, but provide guidance that the Management Plugin must be enabled.

Reply:
Will it actually fail a test or mark it as skipped?

Reply:
It will fail. Anywhere the `\Magento\TestFramework\MessageQueue\PreconditionFailedException` is caught tests are marked failed. https://github.com/magento/magento2/search?q=PreconditionFailedException&unscoped_q=PreconditionFailedException